great_expectations/expectations/metrics/util.py

```diff
@@ -308,6 +321,42 @@ def attempt_allowing_relative_error(dialect):
     return detected_redshift or detected_psycopg2
 
 
+class _CaseInsensitiveString(str):
+    def __init__(self, string: str):
+        self._original = string
+        self._lower = string.lower()
+        self._quote_string = '"'
+
+    def __eq__(self, other):
+        if self.is_quoted():
+            return self._original == str(other)
+        if isinstance(other, _CaseInsensitiveString):
+            return self._lower == other._lower
+        elif isinstance(other, str):
+            return self._lower == other.lower()
+        else:
+            return False
+
+    def __hash__(self):
+        return hash(self._lower)
+
+    def is_quoted(self):
+        return self._original.startswith(self._quote_string)
+
+
+class SmartColumnLookup(UserDict):
```

Comment:
Rename this since it only does a special get for the key `name`? Maybe `CaseInsensitiveNameDict` though that's a bit awful too.

Reply:
That sounds good to me. If it becomes something else we can rename it.